experimental/pages/demologin/views/demologin_view.dart

```diff
@@ -48,17 +50,21 @@ class DemologinView extends GetView<DemologinController> {
                     onPressed: () async {
                       // バリデーションをパスできた場合
                       if(_formKey.currentState?.saveAndValidate() ?? false) {
-                        //TODO: 認証の判定
-                        if(await Future.value(true)){
-                          Get.offNamed(DemoRoutes.demoHome);
+                        DemoUser? user = await controller.authentication(_emailFieldController.text);
+                        // ユーザーを取得できた場合
+                        if(user != null){
+                          Get.offNamed(DemoRoutes.demoHome, arguments: user); // ホーム画面へ遷移
+                        }
+                        // ユーザーを取得できない場合
+                        else{
+                          _formKey.currentState?.invalidateField(name: _fieldNameEmail, errorText: "メールアドレスが存在しません。");
                         }
```

Comment:
```suggestion
                        // ユーザーを取得できた場合
                          if (user == null) {
                            // ユーザーを取得できない場合
                            _formKey.currentState?.invalidateField(
                                name: _fieldNameEmail,
                                errorText: "メールアドレスが存在しません。");
                            return;
                          }
                          Get.offNamed(DemoRoutes.demoHome,
                              arguments: user); // ホーム画面へ遷移
                          return;
```
if else分多いので早期退出にしましょう

Reply:
修正しました！